### PR TITLE
Changed (IBasicFilter | IAdvancedFilter) to IFilter

### PR DESCRIPTION
**Pull request by wbreza**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ gulp.task('ghpages', 'Deploy documentation to gh-pages', ['nojekyll'], function
 
 gulp.task("docs", 'Compile documentation from src code', function () {
   return gulp
-    .src(["src/**/*.ts"])
+    .src(["typings/globals/es6-promise/index.d.ts", "src/**/*.ts"])
```
Comment (wbreza) [inline gulpfile.js]: This allows the typedoc to generate docs with Promise<T>

Comment (mattmazzola) [inline gulpfile.js]: This is unrelated to changing filter types / interfaces though.  

Would rather fix typedoc in separate PR with all of the needed .d.ts files.  
I assume it needs all the dependent libraries, perhaps just include all typings folder.

Comment (wbreza) [inline gulpfile.js]: Big reason for updating to `IFilter` was to support typedoc.  Typedoc doesn't support the (A|B) syntax.  If we include all the typings then it will generate docs for dev dependencies like jasmine, etc.

Comment (mattmazzola) [inline gulpfile.js]: It sounds like there are two different issues here.

1. Typedoc doesn't support union types
2. External .d.ts files need to be included
(es6-promise, wpmp, hpm, router, models)

The second issue is indepedent
